opencti-platform/opencti-front/src/private/components/common/stix_core_objects/StixCoreObjectFileExport.jsx

```diff
@@ -0,0 +1,265 @@
+import React, { useState } from 'react';
+import { map } from 'ramda';
+import Tooltip from '@mui/material/Tooltip';
+import { FileExportOutline } from 'mdi-material-ui';
+import ToggleButton from '@mui/material/ToggleButton';
+import { DialogTitle } from '@mui/material';
+import Dialog from '@mui/material/Dialog';
+import { Field, Form, Formik } from 'formik';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import Button from '@mui/material/Button';
+import * as Yup from 'yup';
+import MenuItem from '@mui/material/MenuItem';
+import { graphql, usePreloadedQuery } from 'react-relay';
+import * as R from 'ramda';
+import { createSearchParams, useNavigate } from 'react-router-dom-v5-compat';
+import { markingDefinitionsLinesSearchQuery } from '../../settings/marking_definitions/MarkingDefinitionsLines';
+import { fileManagerExportMutation } from '../files/FileManager';
+import useQueryLoading from '../../../../utils/hooks/useQueryLoading';
+import Loader, { LoaderVariant } from '../../../../components/Loader';
+import { fieldSpacingContainerStyle } from '../../../../utils/field';
+import SelectField from '../../../../components/SelectField';
+import { useFormatter } from '../../../../components/i18n';
+import { commitMutation, MESSAGING$, QueryRenderer } from '../../../../relay/environment';
+
+const stixCoreObjectFileExportQuery = graphql`
+  query StixCoreObjectFileExportQuery {
+    connectorsForExport {
+      id
+      name
+      active
+      connector_scope
+      updated_at
+    }
+  }
+`;
+
+// TODO see if possible to extract common logic between this component
+// and StixCoreObjectsExportCreation to limit duplicated code.
+
+const exportValidation = (t) => Yup.object().shape({
+  format: Yup.string().required(t('This field is required')),
+});
+
+export const scopesConn = (exportConnectors) => {
+  const scopes = R.uniq(
+    R.flatten(R.map((c) => c.connector_scope, exportConnectors)),
```

Comment:
Please try as much as possible to not use Ramda :)

Reply:
To be honest, this part was copied from an other component and a mess to understand what is going on in there. That's why we kept it like this

Reply:
We changed it, except for uniq which is very convenient

Reply:
Code removed